plugin/storage/cassandra/factory.go

```diff
@@ -86,3 +103,19 @@ func (f *Factory) CreateSpanWriter() (spanstore.Writer, error) {
 func (f *Factory) CreateDependencyReader() (dependencystore.Reader, error) {
 	return cDepStore.NewDependencyStore(f.primarySession, f.Options.DepStoreDataFrequency, f.metricsFactory, f.logger), nil
 }
+
+// CreateArchiveSpanReader implements storage.ArchiveFactory
+func (f *Factory) CreateArchiveSpanReader() (spanstore.Reader, error) {
+	if f.archiveSession == nil {
+		return nil, storage.ErrArchiveStorageNotConfigured
+	}
+	return cSpanStore.NewSpanReader(f.archiveSession, f.metricsFactory, f.logger), nil
+}
+
+// CreateArchiveSpanWriter implements storage.ArchiveFactory
+func (f *Factory) CreateArchiveSpanWriter() (spanstore.Writer, error) {
+	if f.archiveSession == nil {
+		return nil, storage.ErrArchiveStorageNotConfigured
+	}
+	return cSpanStore.NewSpanWriter(f.archiveSession, f.Options.SpanStoreWriteCacheTTL, f.metricsFactory, f.logger), nil
```

Comment:
I'm a bit confused, is `f.Options.SpanStoreWriteCacheTTL` shared between both the archive and primary span writers?

Reply:
at the moment, yes. The retention period of archive storage is expected to be much longer that for primary, so the short cache TTL for the primary is fine for archive too. Strictly speaking we don't even need this cache for archive, but whatever.